### PR TITLE
Fixes for Tagliatelle and Ports of Weondergem

### PR DESCRIPTION
**Pull request by andyHa**

### CODE REVIEW

File: src/main/resources/default/templates/http/error.html.pasta [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+    <w:pageHeader>
+        @i18n('BasicController.error')
+    </w:pageHeader>
+
+    <div class="well">
+        <pre style="font-size: small">
+            @message
+
+            Disgnos
```
Comment (sabieber) [inline src/main/resources/default/templates/http/error.html.pasta]: woot?

File: src/main/java/sirius/tagliatelle/tags/TagElse.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import sirius.kernel.di.std.Register;
+import sirius.tagliatelle.emitter.CompositeEmitter;
+import sirius.tagliatelle.emitter.Emitter;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Handles <tt>i:else</tt> within an <tt>i:if</tt>.
+ */
+public class TagElse extends TagHandler {
```
Comment (sabieber) [inline src/main/java/sirius/tagliatelle/tags/TagElse.java]: ElseTag?

File: src/main/java/sirius/tagliatelle/tags/TagIf.java [hunk 3]
```diff
@@ -8,13 +8,32 @@
...
+import sirius.kernel.di.std.Register;
 import sirius.tagliatelle.emitter.CompositeEmitter;
 import sirius.tagliatelle.emitter.ConditionalEmitter;
 
+import javax.annotation.Nonnull;
+
 /**
  * Handles <tt>i:if</tt> which emits its body if a condition is met.
  */
 public class TagIf extends TagHandler {
```
Comment (sabieber) [inline src/main/java/sirius/tagliatelle/tags/TagIf.java]: IfTag?

File: src/main/java/sirius/tagliatelle/compiler/Parser.java [hunk 4]
```diff
@@ -389,15 +398,48 @@ private Expression call(Expression self) {
...
         call.bindToMethod(position, context, methodName);
-        consumeExpectedCharacter(')');
 
         return call;
     }
 
     /**
+     * Checks if a method call can be interpreted as cast expression (<tt>.as(ClassLiteral)</tt>).
+     * <p>
+     * To make the syntax a bit more pleasing we to casts as ".as" operation instead of double brackets.
```
Comment (sabieber) [inline src/main/java/sirius/tagliatelle/compiler/Parser.java]: to -> do

File: src/main/java/sirius/web/http/Response.java [hunk 5]
```diff
@@ -265,10 +248,7 @@ private void setupCors(DefaultHttpResponse response) {
             return;
         }
 
-        String requestedOrigin = wc.getHeader(HttpHeaderNames.ORIGIN);
-        if (Strings.isFilled(requestedOrigin)) {
-            response.headers().set(HttpHeaderNames.ACCESS_CONTROL_ALLOW_ORIGIN, requestedOrigin);
-        }
+        response.headers().set(HttpHeaderNames.ACCESS_CONTROL_ALLOW_ORIGIN, "*");
```
Comment (andyHa) [inline src/main/java/sirius/web/http/Response.java]: OH GOTT!!!!

File: src/main/java/sirius/web/http/Response.java [hunk 6]
```diff
@@ -1160,27 +998,68 @@ public void template(String name, Object... params) {
...
+            sendTemplateContent(status, template.getEffectiveFileName(), renderContext.toString());
+        } catch (Exception e) {
+            handleTemplateError(template.getEffectiveFileName(), e);
+        }
+    }
+
+    private void setupContentType(Template template) {
+        String fileName = template.getEffectiveFileName();
+        if (fileName.endsWith(".html")) {
+            setHeader(HttpHeaderNames.CONTENT_TYPE, "text/html; charset=UTF-8");
```
Comment (andyHa) [inline src/main/java/sirius/web/http/Response.java]: use constant?

File: src/main/java/sirius/web/http/Response.java [hunk 7]
```diff
@@ -1350,232 +1166,12 @@ public void tunnel(final String url, @Nullable Consumer<Integer> failureHandler)
             }
 
             // Tunnel it through...
-            brb.execute(new TunnelHandler(url, failureHandler));
+            brb.execute(new TunnelHandler(this, wc, url, failureHandler));
```
Comment (andyHa) [inline src/main/java/sirius/web/http/Response.java]: duplicate parameters!

File: src/main/java/sirius/web/templates/velocity/ImageReplacedElementFactory.java [hunk 8]
```diff
@@ -44,6 +46,16 @@
  */
 class ImageReplacedElementFactory extends ITextReplacedElementFactory {
 
+    private static final String TAG_TYPE_IMG = "img";
+    private static final String ATTR_SRC = "src";
+    private static final String ATTR_TYPE = "type";
+    private static final String PROTOCOL_HTTP = "http";
+    private static final String BARCODE_TYPE_QR = "qr";
+    private static final String BARCODE_TYPE_CODE128 = "code128";
+    private static final String BASRCODE_TYPE_EAN = "ean";
```
Comment (andyHa) [inline src/main/java/sirius/web/templates/velocity/ImageReplacedElementFactory.java]: tzpo

**Pull request merged**